jwst/outlier_detection/outlier_detection.py

```diff
@@ -131,81 +226,94 @@ def build_suffix(self, **pars):
+                np.sum(badmask) / len(weight.flat) * 100))
+            badmasks.append(badmask)
+
+        # Fill resampled_sci array with nan's where mask values are True
+        for f1, f2 in zip(sci, badmasks):
+            for elem1, elem2 in zip(f1, f2):
+                elem1[elem2] = np.nan
+
+        # For a stack of images with "bad" data replaced with Nan
+        # use np.nanmedian to compute the median.
+        with warnings.catch_warnings():
+            warnings.filterwarnings(action="ignore",
+                                    message="All-NaN slice encountered",
+                                    category=RuntimeWarning)
+            median_image = np.nanmedian(sci, axis=0)
 
-        # Perform median combination on set of drizzled mosaics
-        median_model.data = self.create_median(drizzled_models)
-        if pars['save_intermediate_results']:
-            if self.outlierpars.get('asn_id', None) is None:
-                suffix_to_remove = self.resample_suffix
-            else:
-                suffix_to_remove = f"_{self.outlierpars['asn_id']}{self.resample_suffix}"
-            median_model_output_path = self.make_output_path(
-                basepath=median_model.meta.filename.replace(suffix_to_remove, '.fits'),
-                suffix='median')
-            median_model.save(median_model_output_path)
-            log.info(f"Saved model in {median_model_output_path}")
-        else:
-            # since we're not saving intermediate results if the drizzled models
-            # were written to disk, remove them
-            if not pars['in_memory']:
-                for fn in drizzled_models._models:
-                    _remove_file(fn)
+        return median_image
 
-        if pars['resample_data']:
-            # Blot the median image back to recreate each input image specified
-            # in the original input list/ASN/ModelContainer
-            blot_models = self.blot_median(median_model)
 
-        else:
-            # Median image will serve as blot image
-            blot_models = ModelContainer(open_models=False)
-            for i in range(len(self.input_models)):
-                blot_models.append(median_model)
+    def create_median(self, resampled_models: ModelContainer) -> ModelContainer:
+        """Create a median image from the singly resampled images.
 
-        # Perform outlier detection using statistical comparisons between
-        # each original input image and its blotted version of the median image
-        self.detect_outliers(blot_models)
+        Parameters
+        ----------
+        resampled_models : ModelContainer
+            A container of resampled images to combine
 
-        # clean-up (just to be explicit about being finished with
-        # these results)
-        if not pars['save_intermediate_results'] and not pars['in_memory']:
-            for fn in blot_models._models:
-                _remove_file(fn)
```

Comment:
Where is this in the new code? I think leaving this out reintroduces a bug fixed in:
https://github.com/spacetelescope/jwst/pull/8481
tests for it are not yet in main (it looks like they're up for review in: https://github.com/spacetelescope/jwst/pull/8481) I doubt this will cause any test failures.

If I run outlier detection locally (with the default options which should have `save_intermediate_results=False`) I see leftover `_blot` files in the working directory.

Reply:
This got removed by accident; will fix, thanks for catching it!